source/agora/common/TransactionPool.d

```diff
@@ -204,20 +213,26 @@ public class TransactionPool
 
     ***************************************************************************/
 
-    public int opApply (scope int delegate(Hash hash, Transaction tx) dg) @trusted
+    public int opApply (scope int delegate(Hash hash, Transaction tx) dg) @trusted nothrow
     {
-        auto results = this.db.execute("SELECT key, val FROM tx_pool");
-
-        foreach (row; results)
+        try
         {
-            auto hash = *cast(Hash*)row.peek!(ubyte[])(0).ptr;
-            auto tx = deserializeFull!Transaction(row.peek!(ubyte[])(1));
+            auto results = this.db.execute("SELECT key, val FROM tx_pool");
 
-            // break
-            if (auto ret = dg(hash, tx))
-                return ret;
-        }
+            foreach (row; results)
+            {
+                auto hash = *cast(Hash*)row.peek!(ubyte[])(0).ptr;
+                auto tx = deserializeFull!Transaction(row.peek!(ubyte[])(1));
 
+                // break
+                if (auto ret = dg(hash, tx))
+                    return ret;
+            }
+        }
+        catch (Exception ex)
+        {
+            // nothing we can do
```

Comment:
We should never catch exceptions and then ignore them. This is also known as the "pokemon exception handling".

Instead:

- Let's make the delegate nothrow
- try/catch on the `execute` call, but then `assert(0)`.

In the future https://github.com/bpfkorea/agora/issues/836 will have to be looked at again - we will need to assess each `execute` call and see if we can handle any sqlite failures - we may want to retry the statement or abort.